server/src/lib/board.ts

```diff
@@ -125,19 +144,72 @@ export class Board {
    * @param room Room destination room to move to
    * @returns boolean
    */
-  public move(user: User, room: Room): boolean {
+  public move(user: User, room: Hallway): boolean {
+    if (this.isLegalMove(user.location(), room)){
+      user.location().exit;
+      room.enter(user);
+      user.currentLocation = room;
+      this.moveMade = true;
+      return true
+    }
+    console.log("Not a valid move. Please try again.");
     return false;
   }
 
   public isLegalAccusation(user: User, accusedRoom: Room): boolean {
-    return user.location === accusedRoom;
+    // user must be in the accusedRoom
+    if (accusedRoom.contains().indexOf(user) >= -1){
+      return true
+    }
+    return false;
+  }
+
+  public isLegalGuess(user: User, guessedRoom: Room): boolean {
+    // user must be in the accusedRoom
+    if (user.currentLocation = guessedRoom){
+      return true
+    }
+    return false;
   }
 
   public cleanup(): boolean {
     return false;
   }
 
   public gameOver(): void {
+    //currently wincondition in board, but make accusation in user. how to link?
+    this.cleanup;
+    return null;
+  }
+
+  public removePlayer(user: User): void{
+    var index = this.players.indexOf(user);
+    if (index > -1) {
+      this.players.splice(index, 1);
+    }
+    console.log("The player " + user.name + " has been removed from the game for a false accusation. Proceeding to next player's turn");
+    this.nextTurn();
+    return null
+  }
+
+  public userSubmitGuess(currentUser: User, guessUser: User, weapon: Weapon, room: Room): void {
+    //checks if person to right has any of the cards guessUser, weapon, room
+    //could use currentPlayer?
+    if (this.isLegalGuess(currentUser, room)){
+      this.guessMade = true;
+      //next player?
+    }else{
+      console.log("User must be in guessed room. Please resubmit guess.")
```

Comment:
We don't want to log move failures to console, since the client won't get that feedback.  Possibly throw a MovementError?  (Which we'd need a class for).  And then let the Controller handle it.